cmd/auth-handler.go

```diff
@@ -105,7 +105,7 @@ func sumMD5(data []byte) []byte {
 }
 
 // Verify if request has valid AWS Signature Version '4'.
-func isReqAuthenticated(r *http.Request) (s3Error APIErrorCode) {
+func isReqAuthenticated(r *http.Request, validateRegion bool) (s3Error APIErrorCode) {
```

Comment:
request needs to be validated for `us-east-1` region in case of putbucket listbuckets getbucketlocation, hence instead of validateRegion, this function can take a region string directly. For putbucket listbuckets getbucketlocation calls we pass `us-east-1` and for other calls we pass `serverConfig.GetRegion()`